static/js/src/advantage/subscribe/react/components/FreeTrialRadio/FreeTrialRadio.test.tsx

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { add, format } from "date-fns";
+import { formatter } from "../../../renderers/form-renderer";
+import * as usePreview from "../../APICalls/usePreview";
+import { preview } from "../../utils/test/Mocks";
+
+import FreeTrialRadio from "./FreeTrialRadio";
+
+const total = 123456;
+
+jest.spyOn(usePreview, "default").mockImplementation(() => {
+  return {
+    isLoading: false,
+    isError: false,
+    isSuccess: true,
+    data: { ...preview, total: total },
+    error: null,
+  };
+});
+
+it("displays a message explaining the trial if free trial is selected", () => {
+  render(
+    <FreeTrialRadio isUsingFreeTrial={true} setIsUsingFreeTrial={() => {}} />
+  );
+  screen.getByText("Your free trial ends:");
+  screen.getByText(
+    `${format(
+      add(new Date(), {
+        months: 1,
+      }),
+      "dd MMMM yyyy"
+    )} after which time you will be charged ${formatter.format(total / 100)}`
+  );
+});
+
+it("does not display the message if pay now is selected", () => {
+  render(
+    <FreeTrialRadio isUsingFreeTrial={false} setIsUsingFreeTrial={() => {}} />
+  );
+  expect(screen.queryByText("Your free trial ends:")).toBeNull();
```

Comment:
This is fine, but in case you didn't know - there's also the `toBeInTheDocument` matcher - which can be more meaningful
https://testing-library.com/docs/guide-disappearance/#nottobeinthedocument

```suggestion
  expect(screen.queryByText("Your free trial ends:")).not.toBeInTheDocument()
```